mitmproxy/protocol/tls.py

```diff
@@ -433,7 +433,9 @@ def server_sni(self):
         """
         The Server Name Indication we want to send with the next server TLS handshake.
         """
-        if self._custom_server_sni is False:
+        if self.config.options.mode == "reverse":
+            return self.server_conn.address.host
+        elif self._custom_server_sni is False:
```

Comment:
Can we just extend the constructor to accept a `server_sni` argument which sets `self._custom_server_sni` and pass that [here](https://github.com/mitmproxy/mitmproxy/blob/8021427ab919e5301d9cc0e5d6ef28f19dee048b/mitmproxy/proxy/root_context.py#L72)?